### DISCUSSION TITLE
The packaged version of WSL is not supported by windows-version 10.0.19045.2130

### DISCUSSION THREAD

**Discussion by sisteu56** [microsoft/WSL#9107] [General] [answered]
Hello,

I have activated the windows feature WSL and restarted my PC as well.

When I run the wsl command from the powershell, I receive the following message "The packaged version of WSL is not supported by windows-version 10.0.19045.2130". 

This is weird since I have windows-version 22H2 (build 19045.2130) installed, which should support WSL.

How can this be fixed?

**Comment by NotTheDr01ds** [comment, marked as answer]
I could be wrong on this, but I believe "the packaged version" here means the *new* WSL release that is only available for Windows 11.  You shouldn't have been able to install it on Windows 10 (even 22H2), but did you maybe find a way to force it somehow?

Open a PowerShell terminal as Admin and try the following:

```
Get-AppxPackage -AllUsers |? { $_.Name -like "*WindowsSubsystemForLinux*" }
```

If that returns any results, then you have the "packaged" version of WSL installed.  I'd be curious how you did it on Windows 10!

But if it returns a result, then try uninstalling it using one of the following methods:

* *Add or Remove Programs* in Settings
* Just find "Windows Subsystem for Linux Preview" in the Start menu, right click, and uninstall
* Assuming that your search above found one-and-only-one result:

    ```
    Get-AppxPackage -AllUsers |? { $_.Name -like "*WindowsSubsystemForLinux*" } | Remove-AppxPackage -AllUsers
    ```

**Reply by Biswa96** [reply]
> I'd be curious how you did it on Windows 10!

See https://github.com/microsoft/WSL/issues/7940#issuecomment-1287074750

**Reply by NotTheDr01ds** [reply]
Yikes!  Good catch.

I'm assuming that you haven't reported this as a separate issue yet, right?  Could you?

**Reply by cedrickcw23** [reply]
@NotTheDr01ds I encountered the same problem, but my issue was installing msi package. When I clicked the installer the installation page did not show up, but it somehow installed all the assets. I found the directory of the new WSL and ran the wsl.exe from the command line and powershell and it said the exact same thing as the questioner. I also tried to delete the wsl folder, but to no avail. 

Thanks for the advice, your explanation helped a lot, and I was able to delete the whole WSL package folder with the command provided. Will use this command if I faced the same issues. At least to ensure a fresh start to installing WSL.

**Comment by Nahor** [comment]
_[Edit] Added mention for older Windows 10_

To clarify:

TLDR; one needs both Win10 22H2 + "22H2 Cumulative Update Preview" to use WSL downloaded from the MS Store.
This might also apply to older versions of Windows 10 but it's unknown if that will work.

Longer description:
- with a basic Win10 22H2 (19045.2130), one can install WSL from the MS store but will get that "not supported" error when trying to start it from there. And when run from the command line (`wsl.exe`), one gets the "old" WSL environment.
- with Win10 22H2 + "22H2 Cumulative Update Preview" (19045.2311), there is no error anymore when run from the store and one gets the "new" `wsl.exe` when using the command line.
- with older versions of Win10 (down to 20H2 / 19042), this may also be true (i.e. won't work with the base Win10, but might work with Cumulative Preview Update).

**Reply by NotTheDr01ds** [reply]
@Nahor  Well, sort of.  To add confusion to your clarification ;-):

* [KB5020030](https://support.microsoft.com/en-gb/topic/november-15-2022-kb5020030-os-builds-19042-2311-19043-2311-19044-2311-and-19045-2311-preview-237a9048-f853-4e29-a3a2-62efdbea95e2), which installs the November cumulative Preview update, is available for all currently supported builds of Windows 10 going back to 19042 (20H2).

* You and I were already upgrade to Windows 10 22H2 before installing KB5020030.  What's not clear is if older Windows 10 versions with KB5020030 installed will work properly with the WSL application package installed from the Store.  I *believe* it should, but I have no way of confirming it short of downloading an older Windows 10 build to install in a VM.

    This likely isn't an issue for most users on their own systems, as we'll have the freedom to update to 22H2.  It's more important for those on corporate-managed systems, who may still be on an older Windows 10 release.  Often these IT groups will still install the monthly servicing releases, so KB5020030 (or later) will probably eventually be installed, but it may be longer before they are updated to 19045 (22H2).

**Reply by Nahor** [reply]
Sure, but then the original post mentions 22H2, so owners of older systems are not as likely to reach this thread either (I found it by searching for the error message).

But the main point for my post was that installing the update preview is one way to fix the problem, and potentially a better solution than uninstalling the packaged WSL if one went through the trouble of installing it rather than using the one coming with Windows in the first place.

I updated my post for older Windows.

**Reply by NotTheDr01ds** [reply]
> Sure, but then the original post mentions 22H2, so owners of older systems are not as likely to reach this thread either (I found it by searching for the error message).

True, too, but that was also a weird situation because KB5020030 wasn't even available when this problem first occurred.  Seems to be a bit of a timing issue with the WSL and Windows Update releases.  20 days ago, I thought for sure the fact that the manifest had been updated to support Windows 10 was a bug in itself -- I'm happy I was wrong!

> I updated my post for older Windows.

Thanks!

**Reply by jslay88** [reply]
and here I am just getting lucky that I happen to be switching over to store WSL and notice I have an update for KB5020030... 😄

**Reply by ArtemFedorov74** [reply]
Куча ошибок которые не исправляються пакетом установщика .msi среди которых : отсутвие строчных значение InstallLocation в разделах \HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\  и HKEY_LOCAL_MACHINE\SOFTWARE\Classes\Installer\Products ; битые  ярлыки относящиеся к продукту в меню пуск.
Разработчики как насчёт того что бы заменить значёк продукта сделав его более информативным , дать в секции ВЕРСИЯ файлов понятные имформативные имена  и дескрипторы ; сделать в установщике .msi выбор локации установки продукта ❓❓❓

**Comment by NotTheDr01ds** [comment]
@sisteu56 I wouldn't mind at all if you are able to change the Accepted Answer to @Nahor's -- It's a better attempt (now that we know what's going on) at answering the actual problem for others who may find it.

I've also added similar information to [this Super User question/answer](https://superuser.com/q/1754138/1210833) for searchability.
